### PR TITLE
add few tests for buffer wrap limit

### PR DESCRIPTION
**Pull request by elshan-eth**
# Description

Add tests to check buffer wrap/unwrap limits.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Optimization: [ ] gas / [ ] bytecode
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [ ] The diff is legible and has no extraneous changes
- [ ] Complex code has been commented, including external interfaces
- [ ] Tests have 100% code coverage
- [ ] The base branch is either `main`, or there's a description of how to merge

### CODE REVIEW

File: pkg/vault/test/foundry/unit/VaultBufferUnit.t.sol [hunk 1]
```diff
@@ -406,6 +414,46 @@ contract VaultBufferUnitTest is BaseVaultTest {
         );
     }
 
+    function testErc4626BufferWrapOrUnwrapWrapExactOutIfAmountInLessThenMin() public {
```
Comment (joaobrunoah) [inline pkg/vault/test/foundry/unit/VaultBufferUnit.t.sol:417]:
```suggestion
    function testWrapExactOutAmountInLessThanMin() public {
```

I'd name it like this, what do you think?

File: pkg/vault/test/foundry/unit/VaultBufferUnit.t.sol [hunk 2]
```diff
@@ -406,6 +414,46 @@ contract VaultBufferUnitTest is BaseVaultTest {
...
+
+        vault.forceUnlock();
+
+        vm.expectRevert(
+            abi.encodeWithSelector(IVaultErrors.WrapAmountTooSmall.selector, IERC4626(address(wDaiInitialized)))
+        );
+        vault.erc4626BufferWrapOrUnwrap(params);
+    }
+
+    function testErc4626BufferUnwrapOrUnwrapWrapExactOutIfAmountInLessThenMin() public {
```
Comment (joaobrunoah) [inline pkg/vault/test/foundry/unit/VaultBufferUnit.t.sol:437]:
```suggestion
    function testUnwrapExactOutAmountInLessThanMin() public {
```

**Review by joaobrunoah — APPROVED** [review]
LGTM! In fact, these tests were missing :)

File: pkg/vault/test/foundry/unit/VaultBufferUnit.t.sol [hunk 3]
```diff
@@ -406,6 +414,46 @@ contract VaultBufferUnitTest is BaseVaultTest {
...
+            direction: WrappingDirection.WRAP,
+            wrappedToken: IERC4626(address(wDaiInitialized)),
+            amountGivenRaw: (_minWrapAmount - 1) * rate,
+            limitRaw: UINT256_MAX
+        });
+
+        vault.forceUnlock();
+
+        vm.expectRevert(
+            abi.encodeWithSelector(IVaultErrors.WrapAmountTooSmall.selector, IERC4626(address(wDaiInitialized)))
```
Comment (jubeira) [inline pkg/vault/test/foundry/unit/VaultBufferUnit.t.sol:432]: I don't see this error used anywhere else. Can we check it for exact in too for consistency?

Comment (elshan-eth) [reply pkg/vault/test/foundry/unit/VaultBufferUnit.t.sol:432]: Do you mean transfer `error WrapAmountTooSmall` to IVault?

**Pull request merged**